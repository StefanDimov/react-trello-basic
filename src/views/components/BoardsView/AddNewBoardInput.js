import React from 'react'
import { FormGroup, FormControl } from 'react-bootstrap'

/**
 * Component with an input for adding a new board. Mean to be used in BoardItemsList.
 * @class comp/AddNewBoardInput
 * @param {function} onCreateBoard - callback for on input submit
 */
export default class AddNewBoardInput extends React.Component {
    constructor(props) { // eslint-disable-line
        super(props)

        this.state = { value: '' }

        this.changeValueHandler = this.changeValueHandler.bind(this)
        this.onSubmit = this.onSubmit.bind(this)
    }

    /**
     * Change handler set on the input
     * @private
     * @param {object} e event object provided to handler
     */
    changeValueHandler(e) {
        this.setState({ value: e.target.value })
    }

    /**
     * Form submit handler. Calls callback with input value.
     * @private
     * @param {object} e event object provided to handler
     */
    onSubmit(e) {
        e.preventDefault()
        this.props.onCreateBoard(this.state.value)
        this.setState({ value: '' })
    }

    render() { // eslint-disable-line
        return (
            <form className="my-add-new-board" onSubmit={this.onSubmit}>
                <FormGroup>
                    <FormControl
                        type="text"
                        value={this.state.value}
                        onChange={this.changeValueHandler}
                        placeholder="Add new board..." />
                </FormGroup>
            </form>
        )
    }
}

AddNewBoardInput.propTypes = {
    onCreateBoard: React.PropTypes.func.isRequired
}
