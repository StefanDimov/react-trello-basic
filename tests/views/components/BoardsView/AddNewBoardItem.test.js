import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { FormControl } from 'react-bootstrap'

import AddNewBoardInput from '../../../../src/views/components/BoardsView/AddNewBoardInput'

describe('AddNewBoardInput', () => {

    const inputChangeEventDataWithName = { target: { value: 'list name' } }
    const onCreateBoard = jest.fn()

    beforeEach(() => {
        onCreateBoard.mockClear()
    })

    it('should render correctly', () => {
        const tree = renderer.create(
            <AddNewBoardInput onCreateBoard={onCreateBoard} />
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('should call onCreateBoard with the new list\'s name on submit', () => {
        const eventMock = { preventDefault: jest.fn() }

        const wrapper = shallow(<AddNewBoardInput onCreateBoard={onCreateBoard} />)

        wrapper.find(FormControl).simulate('change', inputChangeEventDataWithName)
        wrapper.simulate('submit', eventMock)

        expect(onCreateBoard).toHaveBeenLastCalledWith(inputChangeEventDataWithName.target.value)
        expect(eventMock.preventDefault).toHaveBeenCalled()
    })

    it('should update state on change', () => {
        const wrapper = shallow(<AddNewBoardInput onCreateBoard={onCreateBoard} />)
        wrapper.find(FormControl).simulate('change', inputChangeEventDataWithName)
        expect(wrapper.find(FormControl).prop('value')).toBe(inputChangeEventDataWithName.target.value)
    })

})
