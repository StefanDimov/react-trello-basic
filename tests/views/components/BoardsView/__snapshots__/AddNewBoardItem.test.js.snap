exports[`AddNewBoardItem should render correctly 1`] = `
<form
  className="my-add-new-board"
  onSubmit={[Function]}>
  <div
    className="form-group">
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="Add new board..."
      type="text"
      value="" />
  </div>
</form>
`;
